force-app/main/default/classes/TestDataFactory.cls

```diff
@@ -0,0 +1,120 @@
+/**
+ * TestDataFactory
+ * 
+ * This utility class generates test data for use in the methods designed to thouroughly test the Lead Trigger.
+ * It provides methods to create leads, contacts, accounts, and other relevant records,
+ * ensuring consistent and reusable test data setup.
+ * 
+ * Each method is designed to produce data with specific characteristics or to
+ * support different test scenarios.
+ */
+public with sharing class TestDataFactory {
+    
+    /**
+     * Method: createLeadData
+     * 
+     * Creates a list of 9 Lead records with varying titles, industries, lead sources, 
+     * and contact details. This method provides a diverse set of leads to test
+     * different lead scenarios, including scoring and title normalization.
+     * 
+     * @return List<Lead> - A list of leads with various attributes for testing.
+     */
+    public static List<Lead> createLeadData(){
```

Comment:
Make this method more dynamic so you can input the number of leads you want generated.